src/pumipic_adjacency.tpp

```diff
@@ -544,5 +548,42 @@ namespace pumipic {
     Kokkos::Profiling::popRegion();
     return found;
   }
+
+  template <typename ParticleType, typename Segment3d>
+  struct NativeParticleHandlerAtElemBdry {
+    NativeParticleHandlerAtElemBdry () {}
+
+    void operator()(o::Mesh& mesh, ParticleStructure<ParticleType>* ptcls,
+                    o::Write<o::LO>& elem_ids, o::Write<o::LO>& inter_faces,
+                    o::Write<o::LO>& lastExit, o::Write<o::Real>& inter_points,
```

Comment:
`inter_faces` and `inter_points` should be member data in the functor not in `operator()` since it is not data that the search itself needs, but only specific to what happens at element intersections.